beep/tests/test_structure.py

```diff
@@ -567,6 +579,23 @@ def test_from_raw_cycler_run_maccor(self):
         # Ensure barcode/protocol are passed
         self.assertEqual(pcycler_run.barcode, "EXP")
         self.assertEqual(pcycler_run.protocol, "xTESLADIAG_000020_CH71.000")
+        steps = pcycler_run.cycles_interpolated.step_type.unique().tolist()
+        # Ensure that charge and discharge steps are processed
+        self.assertEqual(steps, ['discharge', 'charge'])
+
+        min_index = pcycler_run.cycles_interpolated.cycle_index.min()
+        if 'step_type' in pcycler_run.cycles_interpolated.columns:
+            discharge_interpolated = pcycler_run.cycles_interpolated[
+                (pcycler_run.cycles_interpolated.step_type == 'discharge')]
+            min_index_df = pcycler_run.cycles_interpolated[(pcycler_run.cycles_interpolated.cycle_index == min_index) &
+                                                           (pcycler_run.cycles_interpolated.step_type == 'discharge')]
+        else:
+            discharge_interpolated = pcycler_run.cycles_interpolated
+            min_index_df = pcycler_run.cycles_interpolated[(pcycler_run.cycles_interpolated.cycle_index == min_index)]
+        matches = discharge_interpolated.groupby("cycle_index").apply(
+            lambda x: np.allclose(x.voltage.values, min_index_df.voltage.values))
+        if not np.all(matches):
+            raise ValueError("cycles_interpolated are not uniform")
```

Comment:
Can you help me understand what functionality is being tested by lines 586-598?. It seems to reproduce the `step_type` based decision logic that is also written into `structure`. Are you implicitly testing if this decision logic is being executed inside `structure`?

Reply:
It is testing if the interpolation has been applied uniformly, effectively moving that check from the class to the test. This PR removes that check in the initialization of the ProcessedCyclerRun class since it's not really relevant now. Also, given our need to reduce the memory and compute time for structuring, it seems like something that can be dropped.